handler/datasource/inspect/itemdeals.go

```diff
@@ -0,0 +1,39 @@
+package inspect
+
+import (
+	"github.com/data-preservation-programs/singularity/model"
+	"gorm.io/gorm"
+)
+
+func GetFileDealsHandler(
+	db *gorm.DB,
+	id uint64,
+) ([]model.Deal, error) {
+	return getFileDealsHandler(db, id)
+}
+
+// @Summary Get all deals that have been made for a file
+// @Tags Data Source
+// @Accept json
+// @Produce json
+// @Param id path string true "Source ID"
+// @Success 200 {array} model.Deal
+// @Failure 500 {object} api.HTTPError
+// @Router /file/{id}/deals [get]
+func getFileDealsHandler(
```

Comment:
I'd suggest adding a test to app_test.go so the below query can be tested against all supported database backend. (You don't even need any real data in the database, just run the CLI command - I see you don't have a CLI command for this and I think it will be useful to create one anyway.)
The reason is some of them may expect slightly different syntax, for example, require quote around field name or table name. 
The github CI will run app_test.go against all database backend, if you want to test it locally, you need to setup a local mysql and postgres using below credentials and db name
https://github.com/data-preservation-programs/singularity/blob/ca63e1282650b0ecc72e40f59ffc747fc67a5546/cmd/app_test.go#L70

Reply:
@xinaxu I will have @elijaharita do this as a follow on ticket unless you object.